gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/ClouderaManagerServiceDiscovery.java

```diff
@@ -105,6 +116,34 @@ private DiscoveryApiClient getClient(ServiceDiscoveryConfig discoveryConfig) {
     return client;
   }
 
+  /**
+   * Get the ClouderaManager configuration change monitor from the associated gateway service.
+   */
+  private ClouderaManagerClusterConfigurationMonitor getConfigurationChangeMonitor() {
```

Comment:
I thought GatewayServices is already a public static object that is accessible from GatewayServer or elsewhere. I don't think we need to go through all this effort to try to use reflection here.